marklogic-client-api/src/test/java/com/marklogic/client/test/StructuredSearchTest.java

```diff
@@ -15,168 +15,240 @@
+
+		SearchHandle sh = queryMgr.search(rawDef, new SearchHandle());
+
+		MatchDocumentSummary[] summaries = sh.getMatchResults();
+		assertNotNull(summaries);
+		assertEquals(1, summaries.length);
+
+		MatchDocumentSummary matchResult = summaries[0];
+		Document metadata = matchResult.getMetadata();
+		Element subKey = (Element)
+			metadata.getElementsByTagNameNS("http://marklogic.com/xdmp/json", "subKey").item(0);
+		assertEquals("string", subKey.getAttribute("type"));
+		assertEquals("sub value", subKey.getTextContent());
+
+		String docStr = Common.testDocumentToString(metadata);
+		String handleStr = matchResult.getMetadata(new StringHandle()).get();
+		assertXMLEqual("Different metadata for handle", docStr, handleStr);
+
+		Document snippet = matchResult.getSnippets()[0];
+		docStr = Common.testDocumentToString(snippet);
+		handleStr = matchResult.getSnippetIterator(new StringHandle()).next().get();
+		assertXMLEqual("Different snippet for handle", docStr, handleStr);
+	}
+
+	@Test
+	void andNotQuery() {
+		QueryManager queryManager = Common.client.newQueryManager();
+
+		// Write the same doc in 2 different collections.
+		XMLDocumentManager mgr = Common.client.newXMLDocumentManager();
+		DocumentWriteSet writeSet = mgr.newWriteSet();
+		writeSet.add("/1.xml", new DocumentMetadataHandle()
+			.withPermission("rest-reader", DocumentMetadataHandle.Capability.READ, DocumentMetadataHandle.Capability.UPDATE)
+			.withCollections("test1"), new StringHandle("<hello>world</hello>"));
+		writeSet.add("/2.xml", new DocumentMetadataHandle()
+			.withPermission("rest-reader", DocumentMetadataHandle.Capability.READ, DocumentMetadataHandle.Capability.UPDATE)
+			.withCollections("test2"), new StringHandle("<hello>world</hello>"));
+		mgr.write(writeSet);
+
+		// Verify a structured query works
+		StructuredQueryBuilder queryBuilder = queryManager.newStructuredQueryBuilder();
+		StructuredQueryDefinition structuredQuery = queryBuilder.and(
+			queryBuilder.term("world"),
+			queryBuilder.andNot(
+				queryBuilder.collection("test1"),
+				queryBuilder.collection("test2")
+			)
+		);
+
+		long count = queryManager.search(structuredQuery, new SearchHandle()).getTotalResults();
+		assertEquals(1, count);
+
+		// Verify a combined query works.
+		String combinedQuery = "<search xmlns=\"http://marklogic.com/appservices/search\">\n" +
+			"  <query>\n" +
+			"    <and-query>\n" +
+			"      <term-query>\n" +
+			"        <text>world</text>\n" +
+			"      </term-query>\n" +
+			"      <and-not-query>\n" +
+			"        <positive-query>\n" +
+			"          <collection-query>\n" +
+			"            <uri>test1</uri>\n" +
+			"          </collection-query>\n" +
+			"        </positive-query>\n" +
+			"        <negative-query>\n" +
+			"          <collection-query>\n" +
+			"            <uri>test2</uri>\n" +
+			"          </collection-query>\n" +
+			"        </negative-query>\n" +
+			"      </and-not-query>\n" +
+			"    </and-query>\n" +
```

Comment:
Wouldn't we expect an and-not-query to be:


			"      <and-not-query>\n" +
			"        <positive-query>\n" +
            "          <and-query>\n" +
			"            <term-query>\n" +
			"              <text>world</text>\n" +
			"            </term-query>\n" +
			"            <collection-query>\n" +
			"              <uri>test1</uri>\n" +
			"            </collection-query>\n" +
			"          </and-query>\n" +
			"        </positive-query>\n" +
			"        <negative-query>\n" +
			"          <collection-query>\n" +
			"            <uri>test2</uri>\n" +
			"          </collection-query>\n" +
			"        </negative-query>\n" +
			"      </and-not-query>\n" +

Reply:
It could be like that, I was just modeling it after the query posted in #1640 . I think the details of the query don't matter (unless it's a specific query that is not working in conjunction with an and-not-query), just need to verify that the and-not-query is being honored in a combined query with and without DMSDK.